Examples/Io/Root/src/RootNTupleReader.cpp

```diff
@@ -0,0 +1,417 @@
+// tracks_numPix1L      | std::vector<int32_t>     | AsJagged(AsDtype('>i4'), he...
+// tracks_numPix2L      | std::vector<int32_t>     | AsJagged(AsDtype('>i4'), he...
+// jet_pt               | std::vector<float>       | AsJagged(AsDtype('>f4'), he...
+// jet_eta              | std::vector<float>       | AsJagged(AsDtype('>f4'), he...
+// jet_phi              | std::vector<float>       | AsJagged(AsDtype('>f4'), he...
+// jet_m                | std::vector<float>       | AsJagged(AsDtype('>f4'), he...
+// jet_q                | std::vector<float>       | AsJagged(AsDtype('>f4'), he...
+// jet_ptmatched_pt     | std::vector<float>       | AsJagged(AsDtype('>f4'), he...
+// jet_ptmatched_eta    | std::vector<float>       | AsJagged(AsDtype('>f4'), he...
+// jet_ptmatched_phi    | std::vector<float>       | AsJagged(AsDtype('>f4'), he...
+// jet_ptmatched_m      | std::vector<float>       | AsJagged(AsDtype('>f4'), he...
+// jet_drmatched_pt     | std::vector<float>       | AsJagged(AsDtype('>f4'), he...
+// jet_drmatched_eta    | std::vector<float>       | AsJagged(AsDtype('>f4'), he...
+// jet_drmatched_phi    | std::vector<float>       | AsJagged(AsDtype('>f4'), he...
+// jet_drmatched_m      | std::vector<float>       | AsJagged(AsDtype('>f4'), he...
+// jet_isPU             | std::vector<int32_t>     | AsJagged(AsDtype('>i4'), he...
+// jet_isHS             | std::vector<int32_t>     | AsJagged(AsDtype('>i4'), he...
+// jet_label            | std::vector<int32_t>     | AsJagged(AsDtype('>i4'), he...
+// recovertex_x         | std::vector<float>       | AsJagged(AsDtype('>f4'), he...
+// recovertex_y         | std::vector<float>       | AsJagged(AsDtype('>f4'), he...
+// recovertex_z         | std::vector<float>       | AsJagged(AsDtype('>f4'), he...
+// recovertex_sumPt2    | std::vector<float>       | AsJagged(AsDtype('>f4'), he...
+// recovertex_isPU      | std::vector<int32_t>     | AsJagged(AsDtype('>i4'), he...
+// recovertex_isHS      | std::vector<int32_t>     | AsJagged(AsDtype('>i4'), he...
+// truthvertex_x        | std::vector<float>       | AsJagged(AsDtype('>f4'), he...
+// truthvertex_y        | std::vector<float>       | AsJagged(AsDtype('>f4'), he...
+// truthvertex_z        | std::vector<float>       | AsJagged(AsDtype('>f4'), he...
+// truthvertex_t        | std::vector<float>       | AsJagged(AsDtype('>f4'), he...
+// truthvertex_isPU     | std::vector<int32_t>     | AsJagged(AsDtype('>i4'), he...
+// truthvertex_isHS     | std::vector<int32_t>     | AsJagged(AsDtype('>i4'), he...
+// jet_tracks_idx       | std::vector<std::vect... | AsObjects(AsVector(True, As...
+// recovertex_tracks... | std::vector<std::vect... | AsObjects(AsVector(True, As...
+// truthvertex_track... | std::vector<std::vect... | AsObjects(AsVector(True, As...
+// clang-format on
+
+ActsExamples::RootNTupleReader::BranchPointerWrapper::BranchPointerWrapper() {
+  m_track_d0 = new std::vector<float>;
```

Comment:
Is there a reason why we need to allocate these on the heap? If so, I would strongly argue to use `std::unique_ptr`.
However, I don't see a reason why these vectors cannot live on the stack, if this is possible, that would be the preferable solution.

Reply:
I think this code is useless. even if we use pointers we do not need to initialize them with an empty vector